src/plugins/rule/schema.js

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Block, Text } from 'slate'
+
+const TYPES = ['rule', 'theorem', 'lemma', 'corollary', 'law', 'proposition']
+
+function normalizeRule(editor, error) {
+    const { code, child, key, node, index } = error
+
+    switch (code) {
+    // A child of different type was expected.
+    case 'child_type_invalid': {
+        // A text child could only have been added to an exercise by Slate in
+        // event that all other children were removed. In such case we just want
+        // to drop the exercise.
+        if (Text.isText(child)) {
+            editor.removeNodeByKey(node.key)
+            return
+        }
+        // A non-empty rule without a statement should get an empty statement.
+        if (index === 0) {
```

Comment:
Since there can be a `title` at 0 index, `statement` can be missing also with `index === 1`, but I think it will be called with `child_min_invalid` error

Reply:
You're right, that case wasn't handled. I've fixed it and added a new test case.